packages/migrations/src/MigrationStorage.ts

```diff
@@ -70,8 +71,13 @@ export class MigrationStorage {
   }
 
   protected getMigrationName(name: string) {
-    // strip extension
-    return name.replace(/\.\w+$/, '');
+    const parsedName = path.parse(name);
+    if (['.js', '.ts'].includes(parsedName.ext)) {
+      // strip extension
+      return parsedName.name;
+    }
```

Comment:
An alternative should you want to avoid the usage of `path`.
```suggestion
    if (name.endsWith('.ts') || name.endsWith('.js')) {
      // strip extension
      return name.substring(0, name.length - 3);
    }
```